Planning process update (#1272)
FlowFuse/website: src/handbook/development/releases/planning.md

# Planning

This page provides a comprehensive overview of the planning processes and procedures that guide our product development at FlowFuse. As a continuously evolving software, FlowFuse employs a structured planning method to ensure timely releases, both for the FlowFuse Cloud and for our self-hosting users.

> Instructions for creating a NPM package release are [here](./process.md).

Reviewer: ```suggestion
```
I think this is a bit off-topic to divert to a different page.

Our planning process is continuously evolving as we find the best way to accommodate both a growing team and an expanding set of requirements for how and what we deliver.

## Cadence

FlowFuse is continuously released to FlowFuse Cloud, and every four weeks, on a Thursday, it is packaged for users who are self-hosting FlowFuse. 

### Milestones

> We track packages using GitHub Milestones. We utilize GitHub issues for planning work. Issues are continuously scheduled by the PM. Releases are structured following the [Semantic Versioning](https://semver.org/#semantic-versioning-200).

Reviewer: ```suggestion
We track releases using GitHub Milestones. We utilize GitHub issues for planning work. Issues are continuously scheduled by the PM. Releases are structured following the [Semantic Versioning](https://semver.org/#semantic-versioning-200).
```

I think in this context 'releases' is a better phrase.

> Milestones indicate the week of the release instead of a Patch version. This leads to a specific structure for Milestones: **MAJORversion.MINORversion.WEEK**. For example, 1.14.2 would represent the second week of the release of 1.14. This method ensures that Milestones can be used as deadlines while allowing continuous planning.

Reviewer: As mentioned when we discussed, I'm not 100% on this bit. Given a prioritised backlog, the expectation is everyone is working on the next most important task assigned to them and they get delivered in that order when they are ready.

It's perhaps as simple as calling it a 'target' not a 'deadline' - with an expectation that, in agile, things happen and targets move. 'deadline' feels too final.

Author: Yes, I'll change it to target.

Author: ```suggestion
Milestones indicate the week of the release instead of a Patch version. This leads to a specific structure for Milestones: **MAJORversion.MINORversion.WEEK**. For example, 1.14.2 would represent the second week of the release of 1.14. This method ensures that Milestones can be used as targets while allowing continuous planning.
```

## Retrospective/Kick-Off

The day after each FlowFuse NPM release, always a Friday, a meeting is scheduled
by the CTO. This meeting includes 2 parts:

 1. A retrospective on the last release. This should capture feedback on what
    went well and what could be improved and generate action items to act on the
    feedback.
 2. Kicking off the next release. The involves:

>     1. The Product Manager describing the themes, goals and priorities of the upcomming issues

Reviewer: ```suggestion
    1. The Product Manager describing the themes, goals and priorities of the upcoming issues
```
In a continuous model, there isn't a break between one release and the next - there is no kick-off. Most team members would already be working on items for the next release. Socialising them on this Friday meeting could be too late. Not sure I have an alternative to suggest - we don't want more meetings. So perhaps its more a forward-looking thing to highlight things coming up beyond the immedate 'up next' items.

Author: > So perhaps its more a forward-looking thing to highlight things coming up beyond the immedate 'up next' items.

Yes right

Each feature demo is recorded by the developer responsible for the feature, detailing its functions and operations. If problems, challenges, or improvement ideas arise during the review of the demo, everyone is encouraged to post a comment under the video within the Slack channel. Additionally, a corresponding GitHub issue should be opened to track and prioritize these points for further discussion and action.

> Demos must be done in good time as ealry as possbile to allow for any follow-up action.

Reviewer: ```suggestion
Demos must be done in good time and as early as possible to allow for any follow-up action.
```
